geodistpy/geodesic.py

```
@@ -27,9 +27,53 @@
+        ValueError: If the string is not a recognised ellipsoid name.
+    """
+    if ellipsoid is None or ellipsoid == "WGS-84":
+        return (WGS84_A, WGS84_F)
+    if isinstance(ellipsoid, str):
+        try:
+            return ELLIPSOIDS[ellipsoid]
+        except KeyError:
+            raise ValueError(
+                f"Unknown ellipsoid '{ellipsoid}'. "
+                f"Available: {list(ELLIPSOIDS.keys())}"
+            )
```

Ruff (0.15.2)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)